### PR TITLE
add sort_by macro method

### PR DESCRIPTION
**Pull request by jreinert**
``` crystal
{{ ["abc", "a", "ab"].sort_by { |string| string.size } }}
```

### CODE REVIEW

**Comment by jreinert** [comment]
@bcardiff done :) Sorry I forgot about that. I was writing a kind of model interface similar to json mapping and was building an initializer that would take arguments from the mapping named tuple. There i needed to sort the arguments by nilable so that args that are mandatory are leftmost.

File: src/compiler/crystal/macros/methods.cr [hunk 1]
```diff
@@ -1889,6 +1889,12 @@ private def intepret_array_or_tuple_method(object, klass, method, args, block, i
     klass.new(object.elements.shuffle)
   when "sort"
     klass.new(object.elements.sort { |x, y| x.interpret_compare(y) })
+  when "sort_by"
+    object.interpret_argless_method(method, args) do
+      raise "reject expects a block" unless block
```
Comment (RX14) [inline src/compiler/crystal/macros/methods.cr]: `sort_by expects a block`?

Comment (jreinert) [inline src/compiler/crystal/macros/methods.cr]: yes

**Comment by straight-shoota** [comment]
@jreinert Could you rebase on master?

**Comment by jreinert** [comment]
@straight-shoota: done!

**Comment by jreinert** [comment]
@RX14 @straight-shoota: ping

**Pull request merged**